app/controllers/products_controller.rb

```diff
@@ -49,4 +53,12 @@ def set_product
     def product_params
       params.require(:product).permit(:title, :description, :price, :image)
     end
+
+    def check_logged_in_user
+      unless logged_in?
```

Comment:
Same logic here with `return if logged_in?`. I also noticed that this method is used in most controller, maybe you could move it to a base_controller and just call it from these controllers if needed.

You will probably have:
```ruby
# controllers/base_controller.rb
class BaseController < ApplicationController

# controllers/procuuts_controller.rb
class ProductsController < BaseController
```

Reply:
Thanks for this suggestion, I was looking for a good solution for this.